### PR TITLE
[JENKINS-71737] fix redirect when submitting cloud changes

### PR DESCRIPTION
**Pull request by car-roll**
See [JENKINS-71737](https://issues.jenkins.io/browse/JENKINS-71737).

#8310 first started as simple redirect fixes, but they never worked for downstream plugins (particularly `ec2` and `azure-vm-agents`). The solution then evolved to make[ a separate rename page that was consistent](https://github.com/jenkinsci/jenkins/pull/8310#issuecomment-1682392118) with the rename logic of `AbstractItem` introduced in #3289 .

Eventually it was discovered that the use of `cloudName` instead of `name` in the downstream plugins, along with a hardcoded workaround in core was what made my initial attempts at redirect fail.

This PR is an alternative to the more complicated #8310 
Fixes the redirect so that `save` goes to the new cloud name page and it removes the apply button from cloud configuration

### Testing done
Tested manually in `ec2` and `azure-vm-agents`
TODO: Create PRs with incremental generated by this PR

### Proposed changelog entries

- Fix redirect when renaming a cloud

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8505"></a>

### CODE REVIEW

**Review by timja — COMMENTED** [review]
doesn't work if someone clicks apply and then makes another change.

(the main reason for the rename page I thought was to avoid the apply button being present)

**Comment by car-roll** [comment]
@timja you are right, in my excitement I forgot about that part 🤦

**Comment by p297343876** [comment]
jjjj

**Comment by timja** [comment]
> @timja you are right, in my excitement I forgot about that part 🤦

Basil suggested on the other PR you could just remove the apply button and that may be the simplest fix and then going with this

File: core/src/main/java/hudson/slaves/Cloud.java
Comment (scherler) [inline core/src/main/java/hudson/slaves/Cloud.java:334]: for what it's worth, I think return FormApply.success("../.."); is the quick fix for the issue. Due to the above underlying url. You will always be in .../configure so you are forced to get back to the overall cloud listing page. If you want some elegance you test whether the name has not changed, then use the old way. However, since index.jelly is e.g. in kubernetes plugin a blank page and doing "." leads to it I would rather be redirected to the overall cloud page, but that is me as a user.

**Comment by scherler** [comment]
https://github.com/scherler/jenkins/tree/JENKINS-71737 is the test that should pass

File: core/src/main/java/hudson/slaves/Cloud.java
Comment (car-roll) [reply core/src/main/java/hudson/slaves/Cloud.java:334]: @scherler I think the main issue with that was that a quick fix was not desired, but rather a proper way to address the UX behavior
See https://github.com/jenkinsci/jenkins/pull/8310#discussion_r1292331498

**Comment by Vlatombe** [comment]
> Basil suggested on the other PR you could just remove the apply button and that may be the simplest fix and then going with this

Can we go with this and forget about this bug?

**Comment by timja** [comment]
> > Basil suggested on the other PR you could just remove the apply button and that may be the simplest fix and then going with this
> 
> Can we go with this and forget about this bug?

Why not just remove the apply button?

**Comment by Vlatombe** [comment]
Yes, this is what I meant.

**Comment by timja** [comment]
> Yes, this is what I meant.

go for it

File: core/src/main/java/jenkins/agents/CloudSet.java
Comment (car-roll) [inline core/src/main/java/jenkins/agents/CloudSet.java:234]: ec2 no longer uses that filed name
https://github.com/jenkinsci/ec2-plugin/pull/892

**Review by timja — APPROVED** [review]
LGTM

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
